src/renderer/mesh_storage.zig

```diff
@@ -859,6 +853,61 @@ pub fn updateMeshes(targetTime: i64) void { // MARK: updateMeshes()
 	}
 }
 
+fn batchUpdateBlocks() void {
+	var lightRefreshList = main.List(*ChunkMesh).init(main.stackAllocator);
+	defer lightRefreshList.deinit();
+
+	var regenerateMesh = std.HashMapUnmanaged(chunk.ChunkPosition, *ChunkMesh, chunk.ChunkPosition.HashContext, 80){};
+	defer regenerateMesh.deinit(main.stackAllocator.allocator);
+
+	// First of all process all the block updates:
+	while(blockUpdateList.dequeue()) |blockUpdate| {
+		const pos = chunk.ChunkPosition{.wx = blockUpdate.x, .wy = blockUpdate.y, .wz = blockUpdate.z, .voxelSize = 1};
+		if(getMeshAndIncreaseRefCount(pos)) |mesh| {
+			const result = mesh.updateBlock(blockUpdate.x, blockUpdate.y, blockUpdate.z, blockUpdate.newBlock, &lightRefreshList);
+			switch(result) {
+				.noChange => {
+					mesh.decreaseRefCount();
+					continue;
+				},
+				.regenerateMesh => {
+					const entry = regenerateMesh.getOrPut(main.stackAllocator.allocator, mesh.pos) catch unreachable;
+					if(entry.found_existing) {
+						mesh.decreaseRefCount();
+					} else {
+						entry.value_ptr.* = mesh;
+					}
+				},
+			}
+		} // TODO: It seems like we simply ignore the block update if we don't have the mesh yet.
+	}
+	{
+		var regenerateMeshIterator = regenerateMesh.iterator();
+		while(regenerateMeshIterator.next()) |entry| {
+			const mesh = entry.value_ptr.*;
+			mesh.generateMesh(&lightRefreshList);
+		}
+	}
+	{
+		// TODO: Don't reschedule light refresh for duplicate meshes.
```

Comment:
I think it's already deduplicated elsewhere, so this probably wouldn't make a difference.